plugins/plugin-chart-echarts/src/utils/series.ts

```diff
@@ -67,25 +70,44 @@ export function formatSeriesName(
   if (typeof name === 'boolean') {
     return name.toString();
   }
-  if (name instanceof Date) {
-    return timeFormatter ? timeFormatter(name) : name.toISOString();
+  if (name instanceof Date || coltype === GenericDataType.TEMPORAL) {
+    const d = name instanceof Date ? name : new Date(name);
+
+    return timeFormatter ? timeFormatter(d) : d.toISOString();
   }
   return name;
 }
 
+export const getColtypesMapping = ({
+  coltypes = [],
+  colnames = [],
+}: {
+  coltypes: GenericDataType[];
+  colnames: string[];
+}): Record<string, number> =>
```

Comment:
As this will be useful for all `ChartDataResponseResult`s, the input values to the function could probably be `ChartDataResponseResult` instead of the individual `coltypes` and `colnames`. Also, the return type should probably be `Record<string, GenericDataType>`

Reply:
done